tools/acl_compressor/sources/main.cpp

```diff
@@ -374,24 +353,53 @@ int main(int argc, char** argv)
 	run_unit_tests();
 #endif
 
-	const Options options = parse_options(argc, argv);
+	Options options;
+
+	if (!parse_options(argc, argv, options))
+	{
+		return -1;
+	}
+
+	if (options.input_filename == nullptr)
+	{
+		printf("An input file is required.\n");
+		return -1;
+	}
 
 	Allocator allocator;
 
-	// Initialize our skeleton
-	RigidSkeleton skeleton(allocator, &clip_01_01::bones[0], clip_01_01::num_bones);
+	std::ifstream t(options.input_filename);
+	std::string str((std::istreambuf_iterator<char>(t)),
+		std::istreambuf_iterator<char>());
+
+	printf("Reading input... ");
+
+	ClipReader reader = ClipReader(allocator, str.c_str(), str.length());
+	if (!reader.read())
+	{
+		ClipReaderError err = reader.get_error();
+		printf("\nError on line %d column %d: %s\n", err.line, err.column, err.get_description());
+		return -1;
+	}
+
+	printf("Done.\n\n");
 
-	// Populate our clip with our raw samples
-	AnimationClip clip(allocator, skeleton, clip_01_01::num_samples, clip_01_01::sample_rate);
-	add_clip_track_data(clip);
+	AnimationClip* clip = reader.get_clip();
+	RigidSkeleton* skeleton = reader.get_skeleton();
 
 	// Compress & Decompress
 	{
-		try_algorithm<UniformlySampledAlgorithm>(options, allocator, clip, skeleton, RotationFormat8::Quat_128);
-		try_algorithm<UniformlySampledAlgorithm>(options, allocator, clip, skeleton, RotationFormat8::Quat_96);
-		try_algorithm<UniformlySampledAlgorithm>(options, allocator, clip, skeleton, RotationFormat8::Quat_48);
+		try_algorithm<UniformlySampledAlgorithm>(options, allocator, *clip, *skeleton, RotationFormat8::Quat_128);
+		try_algorithm<UniformlySampledAlgorithm>(options, allocator, *clip, *skeleton, RotationFormat8::Quat_96);
+		try_algorithm<UniformlySampledAlgorithm>(options, allocator, *clip, *skeleton, RotationFormat8::Quat_48);
 	}
 
+	clip->~AnimationClip();
```

Comment:
in that case, we should clearly transfer the ownership, perhaps with something like unique_ptr. We just need to make sure unique_ptr works well with our custom allocator when it frees things